Date:	Sat, 10 Jan 2009 19:06:07 -0800
From:	Karen Xie <kxie@chelsio.com>
Message-Id: <200901110306.n0B367W7016999@localhost.localdomain>
To:	open-iscsi@googlegroups.com, linux-scsi@vger.kernel.org
Subject: [PATCH v2] iscsi_tcp - make padbuf non-static
Cc:	michaelc@cs.wisc.edu, James.Bottomley@HansenPartnership.com,
	kxie@chelsio.com
Sender:	linux-scsi-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-scsi.vger.kernel.org>
X-Mailing-List:	linux-scsi@vger.kernel.org

[PATCH v2] iscsi_tcp - make padbuf non-static

From: Karen Xie <kxie@chelsio.com>

Rebased patch on scsi-misc tree head.

virt_to_page() call should not be used on kernel text and data addresses.
virt_to_page() is used by sg_init_one(). So change padbuf to be allocated within iscsi_segment.

Signed-off-by: Karen Xie <kxie@chelsio.com>
---

 drivers/scsi/iscsi_tcp.c |    3 +--
 include/scsi/iscsi_tcp.h |    1 +
 2 files changed, 2 insertions(+), 2 deletions(-)


diff --git a/drivers/scsi/iscsi_tcp.c b/drivers/scsi/iscsi_tcp.c
index a745f91..e7705d3 100644
--- a/drivers/scsi/iscsi_tcp.c
+++ b/drivers/scsi/iscsi_tcp.c
@@ -190,7 +190,6 @@ iscsi_tcp_segment_splice_digest(struct iscsi_segment *seg
 static inline int
 iscsi_tcp_segment_done(struct iscsi_segment *segment, int recv, unsigned copied)
 {
-	static unsigned char padbuf[ISCSI_PAD_LEN];
 	struct scatterlist sg;
 	unsigned int pad;
 
@@ -245,7 +244,7 @@ int iscsi_tcp_segment_done(struct iscsi_segment *segment,
 		debug_tcp("consume %d pad bytes\n", pad);
 		segment->total_size += pad;
 		segment->size = pad;
-		segment->data = padbuf;
+		segment->data = segment->padbuf;
 		return 0;
 	}
 
diff --git a/include/scsi/iscsi_tcp.h b/include/scsi/iscsi_tcp.h
index 83e32f6..9e3182e 100644
--- a/drivers/scsi/iscsi_tcp.h
+++ b/drivers/scsi/iscsi_tcp.h
@@ -40,6 +40,7 @@ struct iscsi_segment {
 	unsigned int		total_copied;
 
 	struct hash_desc	*hash;
+	unsigned char		padbuf[ISCSI_PAD_LEN];
 	unsigned char		recv_digest[ISCSI_DIGEST_SIZE];
 	unsigned char		digest[ISCSI_DIGEST_SIZE];
 	unsigned int		digest_len;
